playbooks/operationalizing/how-full-is-my-cluster.adoc

```diff
@@ -0,0 +1,150 @@
+---
+---
+= How Full is my OpenShift Cluster?
+Raffaele Spazzoli <rspazzol@redhat.com>
+v1.0, 2017-08-28
+:toc: macro
+:toc-title:
+
+toc::[]
+
+
+
+= Building a Capacity Management Strategy for OpenShift?
+
+In this article we will lay the foundations to do proper capacity management in OpenShift . +
+But before that we need to make sure we understand the question.
+
+== Quotas, Requests, Limits
+
+https://docs.openshift.com/container-platform/latest/dev_guide/compute_resources.html#dev-quotas[Quotas], https://docs.openshift.com/container-platform/latest/dev_guide/compute_resources.html#dev-compute-resources[requests and limits] all play a role in the way OpenShift allocates resources and it can be easy to get them confused.  They are actually taken in consideration by OpenShift at different time and for different purposes. The following table summarizes these concepts:
+
+|===
+||*When it’s evaluated*|*Purpose*
+
+|Quotas|Request time.|To limit how many resources a single tenant can request so that they cannot take over the cluster.
+|Requests|https://docs.openshift.com/container-platform/latest/admin_guide/scheduling/scheduler.html[Scheduling time].|Requests are used to choose which node is the best fit for the given workload (together with other criteria).
+|Limits|Run time.|Limits are used to limit the maximum amount of resources that a container can use at runtime.
+|===
+
+=== Quotas
+
+
+Quotas are attached to a project and have little to do with the real capacity of the cluster. But if a project reaches its quota, the cluster will not accept any additional requests from it, behaving as if it were full (notice that quotas can be attached to multiple projects with https://docs.openshift.com/container-platform/latest/admin_guide/multiproject_quota.html[multiproject quotas]).  
+
+*Best Practice*: create T-shirt sized projects. By that I mean cluster administrators will define, via templates, a few standard sizes for projects where the size is determined by the associated quotas.
+
+=== Requests
+
+Requests are an estimate of what a container will consume when it runs. OpenShift uses this information to create a map of all the nodes and of how many resources they have available based on the containers that have been already allocated. Here is a simple example:
+
+image::images/HowFullIsMyCluster.png[title=How full is my cluster?]
+
+
+In this example, a new pod which is requesting 2GB of memory needs to be allocated to nodes with already committed memory. OpenShift can place this pod only on node1 because node2 does not have enough available resources. +
+OpenShift can deny scheduling of pods if there is no node that can satisfy the request constraint. +
+We can get an idea of how OpenShift estimates a node being utilized with the following command:  +
+oc describe node <name of a node>
+
+This will produce an output of the form (only the relevant fragments are reported): +
+```
```

Comment:
can we switch this code block to asciidoc format?

Reply:
done